lib/aruco_detector.hpp

```diff
@@ -1,55 +1,69 @@
+#pragma once
+
 #include <opencv2/opencv.hpp>
 #include <opencv2/aruco.hpp>
-#include <memory>
-#include "classification_model.hpp"
+#include <iostream>
+#include <vector>
+
+struct ArUcoDetectionResult {
+    std::vector<int> ids;
+    std::vector<std::vector<cv::Point2f>> corners;
+    std::vector<cv::Vec3d> rvecs;
+    std::vector<cv::Vec3d> tvecs;
+};
 
-class Aruco_Detector : public ClassificationModel::dClassification {
+class Aruco_Detector {
 public:
-    Aruco_Detector(cv::VideoCapture& videoCapture, int dictionaryType) : inputVideo(videoCapture), dictionaryType(dictionaryType) {}
+    Aruco_Detector(const cv::Mat& cameraMatrix, int dictionaryType,)
+            : cameraMatrix(cameraMatrix), dictionaryType(dictionaryType) {}
 
-    ClassificationModel::Detection classify(cv::Mat& image, float threshold) override {
+    enum Detection {
+        NONE = 0,
+        ARUCO
+    };
```

Comment:
Any reason to bring these into this class? They can be accessed using `ClassificationModel::Detection` if the `classification_model.hpp` header is included.